README.md

```diff
@@ -64,6 +64,96 @@ source .venv/bin/activate
 
 For more detailed development setup and optional dependency installation options, see [CONTRIBUTING.md](CONTRIBUTING.md#development-environment-setup-with-uv).
 
+### Option 3: Using Docker
```

Comment:
we maintain a document page: https://zilliztech.github.io/deep-searcher/, so it's better to repeat this additional document to e.g. docs/usage